include/solvers/mpm_explicit.tcc

```diff
@@ -179,18 +182,8 @@ bool mpm::MPMExplicit<Tdim>::solve() {
 #endif
 #endif
 
-    if (step_ % output_steps_ == 0) {
-      // HDF5 outputs
-      this->write_hdf5(this->step_, this->nsteps_);
-#ifdef USE_VTK
-      // VTK outputs
-      this->write_vtk(this->step_, this->nsteps_);
-#endif
-#ifdef USE_PARTIO
-      // Partio outputs
-      this->write_partio(this->step_, this->nsteps_);
-#endif
-    }
+    // Write outputs
+    this->write_outputs(this->step_ + 1);
```

Comment:
@bodhinandach  The way I read this, we will check the `step % this->output_steps_` within the `write_output()` function. So we call the new function every single loop.  

Would there be a meaningful time advantage to having the function `write_output()` after a `step % this->output_steps_` IF statement (similiar to before). Then we will only call the function, say, every 10,000 loops.

Reply:
Actually, I prefer to check the if condition inside the function `write_outputs`. The function is a void, so calling it costs really less. Hanging an if condition is not that elegant IMO.